Chaincase.UI/ViewModels/SendViewModel.cs

```diff
@@ -404,6 +406,43 @@ public async Task<bool> SendTransaction(string password)
             return false;
         }
 
+        private IPayjoinClient? GetPayjoinClient()
+        {
+            if (!string.IsNullOrWhiteSpace(PayjoinEndPoint) &&
+                Uri.IsWellFormedUriString(PayjoinEndPoint, UriKind.Absolute))
+            {
+                var payjoinEndPointUri = new Uri(PayjoinEndPoint);
+                if (!_config.UseTor)
+                {
+                    if (payjoinEndPointUri.DnsSafeHost.EndsWith(".onion", StringComparison.OrdinalIgnoreCase))
+                    {
+                        Logger.LogWarning("PayJoin server is an onion service but Tor is disabled. Ignoring...");
+                        return null;
+                    }
+
+                    if (_config.Network == Network.Main && payjoinEndPointUri.Scheme != Uri.UriSchemeHttps)
+                    {
+                        Logger.LogWarning("PayJoin server is not exposed as an onion service nor https. Ignoring...");
+                        return null;
+                    }
+                }
+
+                return new PayjoinClient(payjoinEndPointUri, _config.TorSocks5EndPoint);
+            }
+
+            return null;
+        }
+
+        public string PayjoinEndPoint
+        {
+            get
+            {
+                string endPoint = null;
+                _destinationUrl.Value?.UnknowParameters?.TryGetValue("pj", out endPoint);
+                return endPoint;
```

Comment:
I believe this transform would be easier to reason about as an ObservableAsPropertyHelper on _destinationUrl's unknown params

Reply:
_destinationUrl is an oaph. this is fine